src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobScheduler.scala

```diff
@@ -429,37 +400,17 @@ class JobScheduler @Inject()(val scheduleHorizon: Period,
             updateJob(job, newJob)
             if (job.softError) processDependencies(jobName, Option(lastErrorTime))
 
-            val clusterPrefix = getClusterPrefix(clusterName)
-
             // Handle failure by either disabling the job and notifying the owner,
             // or just notifying the owner.
             if (disableJob) {
               log.warning("Job failed beyond retries! Job will now be disabled after "
                 + newJob.errorsSinceLastSuccess + " failures (disableAfterFailures=" + disableAfterFailures + ").")
-              message match {
-                case Some(msg) =>
-                  sendNotification(job, "%s [Chronos] JOB DISABLED: '%s'".format(clusterPrefix, job.name),
-                    Some("\nFailed at '%s', %d failures since last success\nTask id: %s\nThe scheduler provided this message:\n\n%s"
-                      .format(DateTime.now(DateTimeZone.UTC), newJob.errorsSinceLastSuccess,
-                        taskId, msg)))
-                case None =>
-                  sendNotification(job, "%s [Chronos] JOB DISABLED: '%s'".format(clusterPrefix, job.name),
-                    Some("\nFailed at '%s', %d failures since last success\nTask id: %s\n"
-                      .format(DateTime.now(DateTimeZone.UTC), newJob.errorsSinceLastSuccess, taskId)))
-              }
+              val msg = "\nFailed at '%s', %d failures since last success\nTask id: %s\n"
+                .format(DateTime.now(DateTimeZone.UTC), newJob.errorsSinceLastSuccess, taskId)
+              jobsObserver.onEvent(JobDisabled(job, TaskUtils.appendSchedulerMessage(msg, taskStatus)))
             } else {
               log.warning("Job failed beyond retries!")
-              message match {
-                case Some(msg) =>
-                  sendNotification(job, "%s [Chronos] job '%s' failed!".format(clusterPrefix, job.name),
-                    Some("\n'%s'. Retries attempted: %d.\nTask id: %s\nThe scheduler provided this message:\n\n%s"
-                      .format(DateTime.now(DateTimeZone.UTC), job.retries,
-                        taskId, msg)))
-                case None =>
-                  sendNotification(job, "%s [Chronos] job '%s' failed!".format(clusterPrefix, job.name),
-                    Some("\n'%s'. Retries attempted: %d.\nTask id: %s\n"
-                      .format(DateTime.now(DateTimeZone.UTC), job.retries, taskId)))
-              }
+              jobsObserver.onEvent(JobRetriesExhausted(job, taskStatus, attempt))
             }
             jobMetrics.updateJobStatus(jobName, success = false)
```

Comment:
We are ignoring taskStatus.getMessage here it seems. We should not remove that logging.